containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -263,7 +263,9 @@ class UnorderedMap {
   //@}
 
  private:
-  enum : size_type { invalid_index = ~static_cast<size_type>(0) };
+  enum : size_type {
+    invalid_index = KOKKOS_IMPL_INVALID_INDEX_TYPE(size_type)
```

Comment:
@dalg24 proposes to use `Kokkos::Experimental::NumericTraits::FINITE_MAX` (see doc, core numerics)

Reply:
```C++
public:
  static constexpr size_type invalid_index = Experimental::finite_max_v<size_type>;

Reply:
Thanks!